### PR TITLE
Prepare RsBanList for JSON API

### PR DESCRIPTION
**Pull request by G10h4ck**

### CODE REVIEW

File: libretroshare/src/services/p3banlist.cc [hunk 1]
```diff
@@ -1304,4 +1299,4 @@ int p3BanList::printBanSources_locked(std::ostream &out)
     return true ;
 }
 
-
+RsBanList::~RsBanList() = default;
```
Comment (csoler) [inline libretroshare/src/services/p3banlist.cc:1302]: In terms of how much size is used and code clarity, I prefer the "virtual ~RsBanList() {}" in the .h: it removes the need to go into the cpp file to figure out what this destructor is actually doing to discover that it does nothing.

**Review by csoler — APPROVED** [review]
Not much to say ;-) See my only comment about the destructor.

File: libretroshare/src/services/p3banlist.cc (hunk 1 above)
Comment (csoler) [reply libretroshare/src/services/p3banlist.cc:1302]: ...and it spares the linking job to the compiler too.

Comment (G10h4ck) [reply libretroshare/src/services/p3banlist.cc:1302]: I do prefer too (`=default` can be used in the header too), but the compiler print a warning about the vtable being emitted in every compilation unit that includes the headed if no method is defined outside the header, so moved the empty destructor in an the implementation file

**Pull request merged**